pkg/csvcopy/csvcopy.go

```diff
@@ -154,21 +153,16 @@ func (c *Copier) Copy(ctx context.Context, reader io.Reader) (Result, error) {
 	counter := &CountReader{Reader: reader}
 	bufferedReader := bufio.NewReaderSize(counter, bufferSize)
 
-	switch {
-	case c.useColumnMapping:
-		if c.skip != 1 {
-			return Result{}, fmt.Errorf("column mapping requires skip to be exactly 1 (one header row)")
+	if c.skip > 0 {
```

Comment:
if skip > 0, we always skip, no matter if we need to parse any header.
After the skip, we then parse if required.